crates/exex/exex/src/notifications.rs

```diff
@@ -21,6 +21,44 @@ pub struct ExExNotifications<P, E> {
     inner: ExExNotificationsInner<P, E>,
 }
 
+/// A trait, that represents a stream of [`ExExNotification`]s. The stream will emit notifications
+/// for all blocks. If the stream is configured with a head via [`ExExNotifications::set_with_head`]
+/// or [`ExExNotifications::with_head`], it will run backfill jobs to catch up to the node head.
+pub trait ExExNotificationsDyn<P, E>:
```

Comment:
I don't like the name for this trait because it's not `Dyn`, it's just a trait. But unsure what would be better, maybe `ExExNotificationsStream`?

Reply:
yep, `ExExNotificationsStream` it's better